src/components/menuAppBar/NavButton.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+
+const NavButton = props => {
+  const router = useRouter();
+  const [isActive, setIsActive] = useState(false);
+  const { name, goTo, route, activeIcon, inActiveIcon } = props;
+
+  useEffect(() => {
+    router.pathname === route ? setIsActive(true) : setIsActive(false);
+  }, [goTo]);
```

Comment:
This doesnt work in this way. Your dependency is function which is recreated on every parent re-render so this will be invoked every time the parent is re-rendered.

Reply:
You should check it over router.pathname.